_posts/EternalFootman/2016-04-06-dictionaries-2.md

```diff
@@ -0,0 +1,19 @@
+
+I understand that exercise was very simple, but that's why I found it so frustrating; when it's just a matter of not having the correct punctuation, problem solving skills are not helpful. I knew how to add a single item to the dictionary, but adding things in a loop was not in the book. I often take too long to learn this sort of basic element to a new topic; I suspect that I'm missing it because it's deemed too easy to spend time on, but my progress on the following content suffers because of this.
+
+<iframe src="https://trinket.io/embed/python3/79eba41245" width="100%" height="356" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+In general, I learned much more from this than I had been with the pair programming approach; this way, I had to learn how each problem worked on my own, and trying to coordinate with a partner was often complicated. In related news, the work was much more difficult.
+The fact that exercises 2 and 3 were reviews of the previous dictionary homework was immensely helpful. Applying the concepts to a new problem helped me understand them, which I had not done very well the first time. For me, this approach (introducing a topic through the reading, going over it in more detail in class, and practicing it for homework) has been the most effective.
+I struggled a great deal with exercises 4 and 5. The material is new enough that I'm not good at figuring out how to solve a problem (I don't have the base knowledge for trial and error to be an option or the experience to know how to find a solution). For 4, I couldn't figure out how to manipulate the numbers without separating them from the keys; I eventaully gave up on it. I had more success with 5, and I do understand how it worked, but I suspect that my results aren't correct (some of the data may have gotten lost over the course of the exercises). I'm less concerned about that than I am about the general concepts and those are there. I think.
+
+I've had some minor trouble with Trinket's Python 3. In addition to the general terror that I'll inadvertently close the window and lose my work, the program occasionally seems to stall out - it sticks in the 'running' stage for all eternity. If this happens, I use the share option to open a new page with the link, and the same code works fine there. I haven't found a consistent bit of code that triggers the issue, but it's ongoing.
+
+Overall, I've had a lot of trouble keeping up since we switched from Turtle. I understand that the goal is to develop problem solving skills, and that does seem like an important long-term skillset - eventually, we should be able to learn a new language on our own, and that seems to be the priority lately. That said, I don't feel at all ready for this step. Trying to learn the new material is difficult enough without having to figure it out on my own, and I know that my quality of work has deteriorated.
```

Comment:
I appreciate the difficulty here.  Stick with me on this- I promise it will be worth it.

Think about what's different about these assignments vs Turtle from the standpoint of what you konw about your program.  WIth Turtle, the program's state is mostly displayed to you, making it easy to learn.  With these programs, it's not.  You have to create the display of info for your brain to hook onto via print statements, and your success is more dependent upon your problem solving technique.  

Once you've figured out what seems to be missing vs Turtle, try to add it back in for yourself.